### PR TITLE
[BugFix] Fix not support GCS authentication in broker load and custom core-site.xml in GCS and Azure

### PR DESCRIPTION
**Pull request by Smith-Cruise**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

In this pr, we do the following things:
1. Fix not supporting GCS authentication in broker load.
2. If the user does not specify credential information in SQL's properties, we will not throw error msgs anymore, use `EmptyCloudConfiguration` instead, so that people can use `core-site.xml` for further configuration. (Only changed for Azure and GCS, AWS is supported already)

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/fs/hdfs/HdfsFsManager.java [hunk 1]
```diff
@@ -732,6 +728,71 @@ public HdfsFs getAzureFileSystem(String path, Map<String, String> loadProperties
         }
     }
 
+    /**
+     * Support for Google Cloud Storage File System
+     * Support gs://
+     *
+     * @param path
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/fs/hdfs/HdfsFsManager.java:735]: I think 735-740 isn't necessary

File: fe/fe-core/src/main/java/com/starrocks/credential/empty/EmptyCloudConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+package com.starrocks.credential.empty;
+
+import com.starrocks.credential.CloudConfiguration;
+import com.starrocks.thrift.TCloudConfiguration;
+import org.apache.hadoop.conf.Configuration;
+
+// If user didn't specific any credential, we create EmptyCloudConfiguration instead.
+// It will use Hadoop default constructor instead, user can put core-site.xml into java CLASSPATH to control
+// authentication manually
+public class EmptyCloudConfiguration implements CloudConfiguration {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/credential/empty/EmptyCloudConfiguration.java:24]:
```suggestion
public static CloudConfiguration EMPTY() {
        return new CloudConfiguration() {
            @Override
            public void toThrift(TCloudConfiguration tCloudConfiguration) {

            }

            @Override
            public void applyToConfiguration(Configuration configuration) {

            }

            @Override
            public String getCredentialString() {
                return "empty";
            }
        };
    }
```
may be you could port it in the CloudConfigurationFactory

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/credential/empty/EmptyCloudConfiguration.java:24]: it could be integrated into tryBuildForStorage()

Comment (Smith-Cruise) [reply fe/fe-core/src/main/java/com/starrocks/credential/empty/EmptyCloudConfiguration.java]: Improved.

File: fe/fe-core/src/main/java/com/starrocks/fs/hdfs/HdfsFsManager.java (hunk 1 above)
Comment (Smith-Cruise) [reply fe/fe-core/src/main/java/com/starrocks/fs/hdfs/HdfsFsManager.java]: deleted

**Review by Youngwb — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Comment by Smith-Cruise** [comment]
@Mergify backport branch-3.0

**Pull request merged**